### PR TITLE
test(migration) add tests for normalizeMigrationDefinition

### PR DESCRIPTION
**Pull request by binoy14**
### Description

Adds test for `normalizeMigrationDefinition` function

### What to review

Not 100% sure if it makes sense and is testing the full capability of the functions, @bjoerge thoughts? 

### Testing

Added :)

### Notes for release

N/A

### CODE REVIEW

**Comment by binoy14** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/sanity-io/sanity/5565?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `next`
  * **PR #5540** <a href="https://app.graphite.dev/github/pr/sanity-io/sanity/5540?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #5566** <a href="https://app.graphite.dev/github/pr/sanity-io/sanity/5566?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #5564** <a href="https://app.graphite.dev/github/pr/sanity-io/sanity/5564?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #5565** <a href="https://app.graphite.dev/github/pr/sanity-io/sanity/5565?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #5542** <a href="https://app.graphite.dev/github/pr/sanity-io/sanity/5542?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #5541** <a href="https://app.graphite.dev/github/pr/sanity-io/sanity/5541?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #5544** <a href="https://app.graphite.dev/github/pr/sanity-io/sanity/5544?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #5550** <a href="https://app.graphite.dev/github/pr/sanity-io/sanity/5550?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #5552** <a href="https://app.graphite.dev/github/pr/sanity-io/sanity/5552?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/sanity-io/sanity/5565?utm_source=stack-comment).

**Review by bjoerge — COMMENTED** [review]
I think it would be nice to assert a couple of more cases:
- Calling the normalized function of node migration hooks resolves to a list of the returned mutations
- Returning nothing/void from an async node migration doesn't include `undefined` in the final list of mutations

File: packages/@sanity/migrate/src/runner/__tests__/normalizeMigrationDefinition.test.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    for await (const item of result(mockAsyncIterableIterator, {} as any)) {
+      res.push(item)
+    }
+
+    expect(res.flat()).toEqual([])
+  })
+})
+
+describe('#createAsyncIterableMutation', () => {
```
Comment (jtpetty) [inline packages/@sanity/migrate/src/runner/__tests__/normalizeMigrationDefinition.test.ts:88]: is this test just testing that we follow an interface?

Comment (binoy14) [reply packages/@sanity/migrate/src/runner/__tests__/normalizeMigrationDefinition.test.ts:88]: for the most part it just ensures that it is returns an iterable. `normalizeMigrateDefinition` internally calls this function so the above tests do actually test most of it's functionality

**Review by jtpetty — APPROVED** [review]

**Pull request merged**